test/integration/dialects/sqlite/dao.test.js

```diff
@@ -72,6 +75,223 @@ if (dialect === 'sqlite') {
       });
     });
 
+    describe('json', function() {
+      it('should tell me that a column is json', function() {
+        return this.sequelize.queryInterface.describeTable('Users')
+          .then(function(table) {
+            expect(table.emergency_contact.type).to.equal('JSON');
+          });
+      });
+
+      it('should stringify json with insert', function() {
+        return this.User.create({
+          username: 'bob',
+          emergency_contact: { name: 'joe', phones: [1337, 42] }
+        }, {
+          fields: ['id', 'username', 'document', 'emergency_contact'],
+          logging: function(sql) {
+            var expected = '\'{"name":"joe","phones":[1337,42]}\'';
+            expect(sql.indexOf(expected)).not.to.equal(-1);
+          }
+        });
+      });
+
+      it('should insert json using a custom field name', function() {
+        var self = this;
+
+        this.UserFields = this.sequelize.define('UserFields', {
+          emergencyContact: { type: DataTypes.JSON, field: 'emergy_contact' }
+        });
+        return this.UserFields.sync({ force: true }).then(function() {
+          return self.UserFields.create({
+            emergencyContact: { name: 'joe', phones: [1337, 42] }
+          }).then(function(user) {
+            expect(user.emergencyContact.name).to.equal('joe');
+          });
+        });
+      });
+
+      it('should update json using a custom field name', function() {
+        var self = this;
```

Comment:
For this, ``this`` reference is required so it cannot be replaced with arrow function..